CMakeLists.txt

```
@@ -17,6 +13,14 @@ find_package(LLVM REQUIRED)
 message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
 message(STATUS "Using LLVMConfig.cmake in ${LLVM_DIR}")
 
+if (LLVM_VERSION_MAJOR LESS 16)
+  set(CMAKE_CXX_STANDARD 14)
+else()
+  set(CMAKE_CXX_STANDARD 17)
+endif()
```

I wonder if we can be more clever about this.  Would be worth testing.

```suggestion
set(CMAKE_CXX_STANDARD $<IF:$<VERSION_LESS:${LLVM_VERSION_MAJOR},16>,14,17>)
```

---

I actually removed those changes in one of the commits since I had to sync with the changes _you_ made -- which rendered my changes moot. I don't think pre-16 compatibility is that important, and the fewer changes needed, the better.